crates/fluvio-spu/src/replication/leader/kv.rs

```diff
@@ -0,0 +1,364 @@
+use std::{sync::Arc, task::Poll, pin::Pin};
+
+use anyhow::Result;
+use fluvio_kv_storage::{Log, LogBasedKVStorage};
+use fluvio_protocol::{
+    record::{RecordSet, RawRecords, Batch, Record},
+    Version,
+};
+use fluvio_storage::{
+    ReplicaStorage,
+    iterators::{FileBatchIterator, FileRecordIterator, RecordItem},
+};
+use futures_util::{Stream, ready, Future, FutureExt};
+use tracing::debug;
+
+use super::{LeaderReplicaState, FollowerNotifier};
+
+const RECORDS_SERIALIZATION_VERSION: Version = 0;
+
+pub type LeaderKVStorage<K, V, S> = LogBasedKVStorage<K, V, LeaderReplicaLog<S>>;
+
+#[derive(Debug)]
+pub struct LeaderReplicaLog<S> {
```

Comment:
Not sure I understand purpose of this.  Can you add more description?

Reply:
Added

Reply:
Can you change this to something like `LeaderReplicaaLogReader` ?  Still confused why this is necessary since that should be in the KV.

Reply:
Why is notifier needed?  Since that would be already done by Leader.

Reply:
`KVStorage` operates on some generic `Log` abstraction. This `LeaderReplicaLog` is an implementation of this abstraction for `LeaderReplicaState`